src/Bonsai.ML.Torch/Diagonal.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.ComponentModel;
+using System.Reactive.Linq;
+using System.Xml.Serialization;
+using static TorchSharp.torch;
+
+namespace Bonsai.ML.Torch
+{
+    /// <summary>
+    /// Creates a diagonal matrix. If input is a 1D tensor, it creates a diagonal matrix with the elements of the tensor on the diagonal.
+    /// If input is a 2D tensor, it returns the diagonal elements as a 1D tensor.
+    /// </summary>
+    [Combinator]
+    [ResetCombinator]
+    [Description("Creates a diagonal matrix with the given data type, size, and value.")]
+    [WorkflowElementCategory(ElementCategory.Source)]
+    public class Diagonal
+    {
+        /// <summary>
+        /// The input matrix.
+        /// </summary>
+        [Description("The input matrix.")]
```

Comment:
```suggestion
        /// Gets or sets the values to include in the diagonal.
        /// </summary>
        [Description("The values to include in the diagonal.")]
```

Rewording for style consistency starting with "Gets or sets" for property documentation. Also it seems this is perhaps not the whole matrix, but rather the values along the diagonal, but I could be wrong in my understanding.